cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/PullAppResponse.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol.operationrunner;
+
+/**
+ * Encapsulates the information generated from a single application pull
+ */
+public class PullAppResponse {
+  private final String appDetailString;
```

Comment:
is there a reason this is a string instead of the actual java class?

Reply:
So if we make it actual `AppRequest` class, will there be unnecessary serialization and deserialization for remote task implementation? Especially when we read the response from `RemoteTaskExecutor`. But nvm looks like all the other task responses are using actual class...